Documentation for HA - Compute Isolation (#1104)
hazelcast/hz-docs: docs/modules/architecture/pages/distributed-computing.adoc

    }

    // ...

    public JobBuilder withConfig(@Nonnull JobConfig jobConfig) {
        this.jobConfig = jobConfig;
        return this;
    }

    // ...

    public JobBuilder withMemberSelector(@Nonnull JetMemberSelector memberSelector) {
        throw new UnsupportedOperationException();
    }

    public JobBuilder asLightJob() {
        isLightJob = true;
        return this;
    }

    public Job start() {
        if (dag != null) return isLightJob ? jet.newLightJob(dag,      jobConfig) : jet.newJob(dag,      jobConfig);
        else             return isLightJob ? jet.newLightJob(pipeline, jobConfig) : jet.newJob(pipeline, jobConfig);
    }    
    
    public Job startIfAbsent() {
        if (isLightJob)  throw new UnsupportedOperationException();
        
        if (dag != null) return jet.newJobIfAbsent(dag,      jobConfig);
        else             return jet.newJobIfAbsent(pipeline, jobConfig);
    }
}
```

Create your `JobBuilder` for the pipeline:

```java
   /**
     * Creates a JobBuilder for a new Jet job with Pipeline definition.
     */
    default JobBuilder newJobBuilder(Pipeline p) {
        return new JobBuilder(this, p);
    }
}

> ```

Reviewer: It should be removed, it's design notes for our implementation :)

Reviewer 2: Yes, the section between:
`Define your `JobBuilder` so that it implements member selection:`
...
`Create your `JobBuilder` for the pipeline:` (and code sample below it)

should be removed

Author: I expected this, but left it in for belts and braces; it has now magically disappeared.

Define your pipeline and any member selection override to submit your Jet job from your Hazelcast Java client:

```java
HazelcastInstance hz = HazelcastClient.newHazelcastClient();
// ... 
Map map = hz.getMap(MAP_NAME);

Pipeline p = Pipeline.create()
                .readFrom(Sources.map(map))
                .map(Entry::getValue)
                .writeTo(sink)
                .getPipeline();

Job job = hz.getJet()
        .newJobBuilder(p)

>         .withMemberSelector(ALL_LITE_MEMBERS)

Reviewer 2: The example description says we want to run on a specific lite member, but we're providing a selector for all lite members

Author: Amended (and updated the code to match previous comment)

        .start();
```

In this form we can clearly identify individual steps taken by the computation:

. Get the Map.
. Read lines from the text source.
. Get the value from the loaded Map.
. Write to the sink.

> . Define the `JobBuilder` for the job.

Reviewer 2: ```suggestion
. Define the job using the `JobBuilder` API.
```

Author: Updated

> . Override the default member settings, and select all lite members.

Reviewer 2: ```suggestion
. Override the default member placement of all cluster members, and select all lite members only.
```

Author: Updated

cluster node to the other. This is the basic principle behind
auto-parallelization and distribution.

> == Word Count with Job Placement  

Reviewer 2: ```suggestion
== Word Count with Job Placement Control
```

Author: Updated

Now we'll take the Word Count task above, but we'll define the location to use for the Jet processing job.

NOTE: Your license key must include `Advanced Compute` to activate this feature.

> We'll use the same `ArrayList` that we used in the previous example., but we'll run the Jet processing job on lite members only.

Reviewer 3: ```suggestion
We'll use the same `ArrayList` that we used in the previous example, but we'll run the Jet processing job on lite members only.
```

Create your `JobBuilder` for the pipeline:

```java
   /**
     * Creates a JobBuilder for a new Jet job with Pipeline definition.
     */
    default JobBuilder newJobBuilder(Pipeline p) {
        return new JobBuilder(this, p);
    }
}
```

Define your pipeline and any member selection override to submit your Jet job from your Hazelcast Java client:

```java
HazelcastInstance hz = HazelcastClient.newHazelcastClient();
// ... 
Map map = hz.getMap(MAP_NAME);

Pipeline p = Pipeline.create()
                .readFrom(Sources.map(map))
                .map(Entry::getValue)
                .writeTo(sink)
                .getPipeline();

Job job = hz.getJet()
        .newJobBuilder(p)
        .withMemberSelector(JetMemberSelector.ALL_LITE_MEMBERS)
        .start();
```

In this form we can clearly identify individual steps taken by the computation:

. Get the Map.
. Read lines from the text source.
. Get the value from the loaded Map.
. Write to the sink.
. Define the job using the `JobBuilder` API.

> . Override the default job placements of all cluster members, and select lite members only.

Reviewer 2: ```suggestion
. Override the default job placement of all cluster members, and select lite members only.
```
I think?

Author: Updated
